pkg/sdn/plugin/proxy.go

```diff
@@ -87,29 +93,42 @@ func (proxy *OsdnProxy) watchEgressNetworkPolicies() {
 	})
 }
 
-func (proxy *OsdnProxy) updateNetworkPolicy(policy osapi.EgressNetworkPolicy) {
-	firewall := make([]proxyFirewallItem, len(policy.Spec.Egress))
+func (proxy *OsdnProxy) updateEgressNetworkPolicy(policy osapi.EgressNetworkPolicy) {
+	if policy.Spec.Egress != nil {
+		if ref, ok := proxy.firewall[policy.Namespace]; ok {
+			if policy.UID != ref.policyUID {
+				// We only allow one policy per namespace otherwise it's hard to determine which policy to apply first
+				glog.Errorf("found multiple egress policies, dropping all firewall rules for namespace: %s", policy.Namespace)
+				delete(proxy.firewall, policy.Namespace)
```

Comment:
Hm... this isn't right, since it means that the next time an ENP in that namespace gets updated, `proxy.firewall[policy.Namespace]` will be unset, and so that ENP *will* get applied (even if there are also still other ENPs in the namespace too).

We need to remember all the ENPs we've seen, but only apply the policy when there's just one in the Namespace.